metadataStore/odm_templates.py

```diff
@@ -26,79 +27,111 @@ class BeginRunEvent(Document):
     owner = StringField(default=getuser(), required=True, unique=False)
-    custom = DictField(unique=False, required=False, default=dict())
-    # Keeping custom a dict for the time being instead of new collection
-
-    meta = {'indexes': ['-_id', '-owner', '-time']}
+    group = StringField(required=False, unique=False, default=None)
+    sample = DictField(required=False)  # lightweight sample placeholder.
+    meta = {'indexes': ['-_id', '-owner', '-time', '-scan_id', '-uid']}
 
 
-class EndRunEvent(Document):
+class EndRunEvent(DynamicDocument):
     """Indicates the end of a series of events
 
     Attributes
     ----------
     begin_run_event : bson.ObjectId
         Foreign key to corresponding BeginRunEvent
-    reason : str
+    exit_status : {'success', 'fail', 'abort'}
         provides information regarding the run success.
-    time : timestamp
+    time : float
         The date/time as found at the client side when an event is
         created.
+    reason : str, optional
+        Long-form description of why the run ended.
+        20 char max.
     """
     time = FloatField(required=True)
     time_as_datetime = DateTimeField()
     begin_run_event = ReferenceField(BeginRunEvent, reverse_delete_rule=DENY,
                                      required=True, db_field='begin_run_id')
 
-    reason = StringField(max_length=10, required=False, default='Complete')
-    custom = DictField(required=False)
-    meta = {'indexes': ['-_id', '-time', '-reason', '-begin_run_event']}
+    exit_status = StringField(max_length=10, required=False, default='success',
+                              choices=('success', 'abort', 'fail'))
+    reason = StringField(required=False, max_length=20)
+    meta = {'indexes': ['-_id', '-time', '-exit_status', '-begin_run_event']}
 
 
-class EventDescriptor(Document):
-    """ Provides information regarding the upcoming series of events
-    whose contents are indicated via such EventDescriptor.
+class DataKeys(DynamicEmbeddedDocument):
+    """Describes the objects in the data property of Event documents.
+    Be aware that this is DynamicEmbeddedDoc Append fields rather than
+    custom dict field
 
     Attributes
     ----------
-    begin_run_event: metadataStore.odm_templates.BeginRunEvent
-        BeginRunEvent object created prior to a BeginRunEvent
-    data_keys : dict
-        e.g.
-        {'key_name' : {'source' : 'PV', 'external' : 'FILESTORE'}}
+    dtype : str
+        The type of data in the event
+    shape : list
+        The shape of the data.  None and empty list mean scalar data.
+    source : str
+        The source (ex piece of hardware) of the data.
+    external : str, optional
+        Where the data is stored if it is stored external to the events.
+    """
+    dtype = StringField(required=True)
+    shape = ListField(required=True, default=None) # defaults to None
+    source = StringField(required=True)
```

Comment:
Can you throw the regexs from the schema document in here?  It looks like `StringField` has an input `regex`